subprojects/plugins/src/main/java/org/gradle/api/tasks/bundling/War.java

```diff
@@ -166,4 +174,17 @@ public void setWebXml(@Nullable File webXml) {
         this.webXml = webXml;
     }
 
+    /**
+     * Returns the app directory of the task. Defaults to {@code src/main/webapp}.
+     *
+     * @return The app directory.
+     * @since 7.1
+     */
+    @Incubating
+    @Optional
```

Comment:
I think `@Optional` doesn't make sense here because webAppDir will never be null and that's really what `@Optional` means